lib/ansible/module_utils/basic.py

```diff
@@ -920,6 +924,20 @@ def __init__(self, argument_spec, bypass_checks=False, no_log=False,
                            ' Update the code for this module  In the future, AnsibleModule will'
                            ' always check for invalid arguments.', version='2.9')
 
+    @property
+    def tmpdir(self):
+        # if _ansible_tmpdir was not set, the module needs to create it and
+        # clean it up once finished.
+        if self._tmpdir is None:
+            basedir = os.path.expanduser(self._remote_tmp)
+            basefile = "ansible-moduletmp-%s-" % time.time()
+            tmpdir = tempfile.mkdtemp(prefix=basefile, dir=basedir)
+            if not self._keep_remote_files:
```

Comment:
I don't think that we want to keep module-created temp files if keep_remote_files is True but I won't block merging on that.  We should talk with nitzmahone about it and possibly remove it in a subsequent PR, though.

Reply:
Hmm... actually, as I think about it, keeping the module-created temporary files may be actively harmful to the use case of keep_remote_files.  Keep_remote_files is there so that someone can reproduce what the module does as closely as possible.  Keeping the module-created temporary directory is a difference for the user when they run the module a second time from when they ran it the first time.  Since we're creating a new tempdir, it shouldn't affect things most of the time but there are cornercases like running the filsystem out of free space.  So I would prefer if we removed the keep_remote_files stuff.